### DISCUSSION TITLE
Robot turns after stopping

### DISCUSSION THREAD

**Discussion by STRG-R** [pybricks/support#1783] [Q&A]
Hi! Im trying to stop my robot after driving forwards. My problem is that I have no idea how go stop it properly. If I use stop() or brake() the robot doesn't stop immediately but goes on for an unpredictable amount of time before stopping (of course not major but the robot has to be precise...). If I use hold() the robot stops precise but the motors roll back a tiny bit after stopping. I think this is because of the tension in the motor that builds while stopping.

And I have another problem: The motors don't stop at the same time. There's a slight delay between the first and the second motor stopping which leads to the robot making a turn. (The lines of code for stopping are directly under each other)

**Comment by laurensvalk** [comment]
Thanks for opening this discussion! To be of more help, it would probably help to see some context or code.

Most movements like `straight` have an automatic, controlled stop, so I suppose you're not using those. I'm going to assume it is already moving using something like `drive()`, such as after line following?

Due to physics, stopping always takes some distance and time.

If you are already using `drive`: This method smoothly accelerates to the given speed. So it could make sense to call `drive(0, 0)`, and then wait for it to come to standstill. For example by checking that the speed becomes small.

**Reply by STRG-R** [reply]
Thanks for your fast response! I'm not using any prebuilt functions. My function basically decelerates from a pretty high speed like 1500 to a lower speed like 250 so stopping is more precise.

**Reply by laurensvalk** [reply]
Are you using the `Motor` functions or the `DriveBase` functions?

> My function basically decelerates from a pretty high speed like 1500 to a lower speed like 250 so stopping is more precise.

Due to hardware limitations, these motors get less accurate at very low speeds (below 250 degrees per second or so), so it is better to avoid driving at very low speeds if you can.

**Reply by STRG-R** [reply]
I'm using the motor functions like motora.run(...) or motora.hold()

Ok interesting... But I have to decelerate if I don't want my wheels to slip... I don't think it's possible to stop perfectly when you have a speed of 1500

**Reply by laurensvalk** [reply]
Which hub are you using?

**Reply by STRG-R** [reply]
Spike Prime
